File: src/algorithms/heaps/operations/heap-peek/__tests__/HeapPeek_test.java
public class HeapPeek_test {
    public static void main(String[] args) {
        assert HeapPeek.heapPeek(new int[]{1,3,5,7,9,8,6}) == 1 : "Test 1 failed";
        assert HeapPeek.heapPeek(new int[]{42}) == 42 : "Test 2 failed";
        assert HeapPeek.heapPeek(new int[]{2,7}) == 2 : "Test 3 failed";
        assert HeapPeek.heapPeek(new int[]{1,3,2,7,5,8,4,9,6}) == 1 : "Test 4 failed";
        int[] heap = {1, 3, 5, 7};
        assert HeapPeek.heapPeek(heap) == HeapPeek.heapPeek(heap) : "Test 5 failed: idempotent";
        System.out.println("All tests passed!");
    }
}

File: src/algorithms/heaps/operations/heap-peek/heap-peek_test.rs
include!("sources/heap-peek.rs");

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_returns_minimum() {
        assert_eq!(heap_peek(&[1,3,5,7,9,8,6]), Some(1));
    }

    #[test]
    fn test_single_element() {
        assert_eq!(heap_peek(&[42]), Some(42));
    }

    #[test]
    fn test_two_element() {
        assert_eq!(heap_peek(&[2,7]), Some(2));
    }

    #[test]
    fn test_idempotent() {
        let heap = &[1,3,5,7];
        assert_eq!(heap_peek(heap), heap_peek(heap));
        assert_eq!(heap_peek(heap), Some(1));
    }

    #[test]
    fn test_larger_heap() {
        assert_eq!(heap_peek(&[1,3,2,7,5,8,4,9,6]), Some(1));
    }
}
